turbinia/task_manager.py

```diff
@@ -95,8 +97,15 @@ def task_runner(obj, *args, **kwargs):
   Returns:
     Output from TurbiniaTask (should be TurbiniaTaskResult).
   """
-  obj = workers.TurbiniaTask.deserialize(obj)
-  return obj.run_wrapper(*args, **kwargs)
+  if os.path.exists(config.LOCK_FILE):
```

Comment:
how about we add a 'or' check to a config.TO_BE_DELETED file as well (created by the preStop hook). If that file exists we psq.Retry as well. That way the preStop can exit without having to worry about a tasks being picked up.

Reply:
Great idea, went ahead and added this in!